src/regdesk/printer_manager.js

```diff
@@ -0,0 +1,66 @@
+import { LabelEpl, LP2844 } from 'WebZLP/src/LP2844'
+
+export class PrinterManager {
+    #devices = [];
+    #printers = [];
+
+    primaryPrinter = 0;
+    secondaryPrinter = 1;
+
+    #printerType = LP2844;
+
+    constructor() {
+    }
+
+    get printers() { return this.#printers; }
+
+    async printerConnect(e) {
+        if (!this.#devices.includes(e.device)) {
+            this.#devices.push(e.device);
+            let printer = new this.#printerType(e.device);
+            await printer.connect();
+            this.#printers.push(printer);
+            this.primaryPrinter = 0;
+        }
+    }
+
+    async printerDisconnect(e) {
+        const idx = this.#devices.findIndex(i => e.device == i);
+        if (idx > -1) {
+            const p = this.#printers[idx];
+            this.#devices.splice(idx, 1);
+            this.#printers.splice(idx, 1);
+            this.resetPrinterButtons();
```

Comment:
I had a huge comment typed up, but after thinking about it this may actually just be a typo.

(If it isn't a typo, oh boy, we got some fun ahead of us :D).

Reply:
Copy/paste error specifically, yep.